manifests/vhost.pp

```diff
@@ -251,8 +251,12 @@
   # This ensures that the docroot exists
   # But enables it to be specified across multiple vhost resources
   if ! defined(File[$docroot]) and $manage_docroot {
+    $docroot_ensure = $ensure? {
+      present => 'directory',
+      default => 'absent',
+    }
```

Comment:
Rather than basing this on ensure could you add a `docroot_ensure` parameter to the define?